tiledb/sm/query/readers/sparse_index_reader_base.cc

```diff
@@ -280,9 +280,14 @@ Status SparseIndexReaderBase::load_initial_data(bool include_coords) {
         read_state_.frag_idx_,
         &result_tile_ranges_));
 
-    for (auto frag_result_tile_ranges : result_tile_ranges_) {
-      memory_used_result_tile_ranges_ += frag_result_tile_ranges.size() *
-                                         sizeof(std::pair<uint64_t, uint64_t>);
+    // Compute the size of the tile ranges structure and mark empty fragments
+    // as fully loaded.
+    for (uint64_t i = 0; i < result_tile_ranges_.size(); i++) {
+      memory_used_result_tile_ranges_ +=
+          result_tile_ranges_[i].size() * sizeof(std::pair<uint64_t, uint64_t>);
+      if (result_tile_ranges_[i].size() == 0) {
+        all_tiles_loaded_[i] = true;
```

Comment:
`result_tile_ranges_.size() == all_tiles_loaded_.size()` because both are initialized w/ `.reserve(fragment_num)`, but it's a bit confusing to need to track that fact down in another class.

- https://github.com/TileDB-Inc/TileDB/blob/3210625773f0ad44092e9c9b2e20d67caab5af0a/tiledb/sm/subarray/subarray.cc#L2426
- https://github.com/TileDB-Inc/TileDB/blob/af79daaf48d0a36d2ad41dd1888b296dd762a861/tiledb/sm/query/readers/sparse_index_reader_base.cc#L265

Reply:
(_absolutely no change needed_ in this bugfix, but noting for future refactoring)